src/v2/models/AppState.js

```diff
@@ -22,69 +21,56 @@ export default Model.extend({
     currentState: 'object',
     factor: 'object',      // optional
     user: 'object',        // optional
-    currentFormName: 'string', // default to first form from Remediation
   },
 
   derived: {
     remediation: {
       deps: ['currentState'],
-      fn: function (currentState = {}) {
+      fn (currentState = {}) {
         return Array.isArray(currentState.remediation) ? currentState.remediation : [];
       },
     },
-    // TODO: change to factorProfile otherwise too many derived props per factor??
-    factorEmail: {
+    factorProfile: {
       deps: ['factor'],
-      fn: function (factor = {}) {
-        return factor.profile && factor.profile.email;
+      fn (factor = {}) {
+        return factor.profile || {};
       },
     },
     factorType: {
       deps: ['factor'],
-      fn: function (factor = {}) {
+      fn (factor = {}) {
         return factor.factorType;
       },
     },
+    isTerminalState: {
+      deps: ['terminal'],
+      fn: function (terminal) {
+        return !_.isEmpty(terminal);
```

Comment:
We not longer get a `step = TERMINAL` right ?

Reply:
no we don't